### PR TITLE
simplify some arbitrary instances in tests

### PR DESCRIPTION
**Pull request by matthewleon**

### CODE REVIEW

File: test/Test/Data/Map.purs [hunk 1]
```diff
@@ -65,19 +66,7 @@ instance ordSmallKey :: Ord SmallKey where
...
-      _ | n < 0.2 -> B
-      _ | n < 0.3 -> C
-      _ | n < 0.4 -> D
-      _ | n < 0.5 -> E
-      _ | n < 0.6 -> F
-      _ | n < 0.7 -> G
-      _ | n < 0.8 -> H
-      _ | n < 0.9 -> I
-      _ -> J
+  arbitrary = oneOf (pure A) (map pure [B, C, D, E, F, G, H, I])
```
Comment (hdgarrood) [inline test/Test/Data/Map.purs]: I think [`elements`](https://pursuit.purescript.org/packages/purescript-quickcheck/3.1.1/docs/Test.QuickCheck.Gen#v:elements) would make this even neater.

Comment (hdgarrood) [inline test/Test/Data/Map.purs]: Also, I think you're missing `J` from the list?

Comment (matthewleon) [inline test/Test/Data/Map.purs]: Good points. Addressed in the next commit. Thanks!

File: test/Test/Data/Map.purs [hunk 2]
```diff
@@ -86,16 +75,7 @@ instance showInstruction :: (Show k, Show v) => Show (Instruction k v) where
...
-    b <- arbitrary
-    case b of
-      true -> do
-        k <- arbitrary
-        v <- arbitrary
-        pure (Insert k v)
-      false -> do
-        k <- arbitrary
-        pure (Delete k)
+  arbitrary = oneOf (Insert <$> arbitrary <*> arbitrary) [(Delete <$> arbitrary)]
```
Comment (hdgarrood) [inline test/Test/Data/Map.purs]: Oops, missed this before - could you please remove the redundant parens around `Delete`?

**Comment by matthewleon** [comment]
from `simplify-test-gens` (new):
```
> import Test.QuickCheck.Arbitrary
> import Test.QuickCheck.Gen
> import Test.Data.Map
> randomSample' 100 (arbitrary :: Gen SmallKey)     [A,F,E,D,D,I,A,C,I,J,D,G,C,B,A,B,C,F,A,H,J,B,B,J,I,A,J,F,J,D,I,J,J,H,I,D,H,E,B,D,B,G,B,H,H,J,C,I,H,F,E,J,G,J,E,C,G,E,D,J,D,I,H,I,F,A,E,G,F,I,B,B,G,A,E,H,D,H,D,A,F,E,E,D,A,D,E,D,I,D,A,D,A,I,B,A,E,B,A,B]
```

from `master` (old)
```
> import Test.QuickCheck.Arbitrary
> import Test.QuickCheck.Gen
> import Test.Data.Map
> randomSample' 100 (arbitrary :: Gen SmallKey)
[J,C,B,I,D,C,F,J,B,E,F,E,J,A,D,E,J,I,D,I,G,F,E,J,A,F,H,H,C,D,E,F,I,A,B,F,J,J,E,F,I,H,C,E,G,A,E,F,C,F,H,I,F,A,C,H,F,C,D,C,E,A,H,H,H,F,A,D,B,A,A,B,A,C,G,B,A,C,E,H,C,F,A,G,H,D,C,D,H,H,A,H,F,D,D,J,I,H,G,I]
```

**Pull request closed without merging**